swap_meet/vendor.py

```diff
@@ -1,2 +1,105 @@
+
+from .item import Item
+
+
+
+# WAVE 1
 class Vendor:
-    pass
+    # constructor that is by default empty, but checks for input
+    def __init__(self, inventory = [] ):
+        if inventory :
+            self.inventory = inventory
+        else:
+            self.inventory = []
+
+    # method thats adds an item to inventory
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    # method that removes an item from inventory
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    # WAVE 2
+    # method that adds items that are the same category to a list
+    def get_by_category(self, category):
+        category_list = []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list
```

Comment:
This would be a great spot to use list comprehension!

```
same_category = [item for item in self.inventory if item.category == category]
```

Here's more examples of Python list comprehensions if you want to continue playing around with them 😄 : https://www.w3schools.com/python/python_lists_comprehension.asp